### ISSUE TITLE
[BUG] Provisioning VM with static IP configuration and vm-dhcp-controller enabled still allocates IP from IPPool

### ISSUE THREAD

**Issue by DovydasNavickas** [harvester/harvester#5452]
**Describe the bug**

Provisioning VM with a static IP set with cloud-init "Network Data" still sends DHCP request to `vm-dhcp-controller` and allocates static IP pair there consuming an IP from the pool. In a case where only a small subset of IPs are given to the IPPool, allocation of the IP is very inconvenient.

**To Reproduce**
1. Enable "vm-dhcp-controler" addon.
2. Setup IPPool for the VM Network.
3. Create a VM and set its cloud-init Network Data to configure network with a static IP address:
```
network:
  version: 1
  config:
    - type: physical
      name: enp1s0
      subnets:
        - type: static
          address: 10.65.1.140/22
          gateway: 10.65.0.1
          dns_nameservers:
            - 10.65.0.1
```

**Expected behavior**

Network interface is configured with a static IP address and no IP is consumed from the `vm-dhcp-controller` IPPool.

**Actual behavior**
DHCP request is still sent to the `vm-dhcp-controller` and even when network interface is configured with a static IP address, there is an IP address consumed from the `vm-dhcp-controller` IPPool and an IP-MAC pair is saved in etcd, even though that IP is never going to be used with a static configuration.

**Support bundle**

Irrelevant.

**Environment**
 - Harvester ISO version: 1.3.0
 - harvester-vm-dhcp-controller version: 0.3.1

**Additional context**
This problem also happens with a multiple network interface configuration as it is described in [KB here](https://harvesterhci.io/kb/multiple-nics-vm-connectivity/).

**Comment by starbops** [comment]
Hi @DovydasNavickas Thanks for looking into the Managed DHCP add-on and giving us feedback! By configuring static IP addresses for the VM via cloud-init network data, the VM should not issue any DHCP requests.

The root cause of this seem-to-be problematic behavior is that, by default, the vm-dhcp-controller keeps an eye on every VirtualMachine object created on the cluster and will, therefore, create corresponding VirtualMachineNetworkConfig objects to try to allocate IP addresses for the VMs' network interfaces. The IP address allocation happens when such objects are created on the control plane, not at the moment the embedded DHCP server receives DHCP requests on the data plane.

So in the current implementation, to avoid IP addresses being allocated from the IP pool is to remove the VirtualMachineNetworkConfig objects associated with the VM.

In the long run, we will have a better integration with Harvester itself, allowing users to choose whether to enable DHCP on a per-VM creation basis.

**Comment by starbops** [comment]
This could be solved once #5052 is delivered. Closed as duplicated.
